packages/device_info_plus/lib/device_info_plus.dart

```diff
@@ -34,20 +34,23 @@ class DeviceInfoPlugin {
   }
 
   static bool _disablePlatformOverride = false;
-  static DeviceInfoPlatform __platform;
+  static DeviceInfoPlatform? __platform;
 
   // This is to manually endorse the Linux plugin until automatic registration
   // of dart plugins is implemented.
   // See https://github.com/flutter/flutter/issues/52267 for more details.
   static DeviceInfoPlatform get _platform {
-    __platform ??= !kIsWeb && Platform.isLinux && !_disablePlatformOverride
+    var platform = __platform;
+
+    platform ??= !kIsWeb && Platform.isLinux && !_disablePlatformOverride
         ? DeviceInfoLinux()
         : DeviceInfoPlatform.instance;
-    return __platform;
+
+    return __platform = platform;
```

Comment:
does this actually need any changes? wouldn't the old assignment work with nullable __platform or did i miss something?

Reply:
We can do it the old way and return `DeviceInfoPlatform?` instead, but if we want return a non-nullable type, the compiler can't guarantee that the class member is not null. Hence the local variable.

Reply:
or return `__platform!`?

Reply:
Ouch, yeah, we could do that as well. I am hesitant to use `!` though, since it's hard to trust the compiler if we do that often. Just a conclusion I've learned from migrating _a lot_ of Java to Kotlin and gotten burnt in runtime.

Reply:
There's an unconditional assignment above so what could possibly go wrong? :)

https://github.com/fluttercommunity/plus_plugins/pull/116/files#diff-1ddc6908e37618bbc4937f9caad0333cfe733aba085bfba4e6e1cf3857271990

Reply:
If that is the preferred way, sure.

Reply:
I found a way that is safe and small :) 
```dart
 return __platform ??= !kIsWeb && Platform.isLinux && !_disablePlatformOverride
        ? DeviceInfoLinux()
        : DeviceInfoPlatform.instance;
```